### PR TITLE
chore(release): kfp-tekton 2.0.5 backend release

### PR DESCRIPTION
**Pull request by Tomcli**
**Which issue is resolved by this Pull Request:** 
Resolves #

**Description of your changes:**

**Environment tested:**

* Python Version (use `python --version`):
* Tekton Version (use `tkn version`):
* Kubernetes Version (use `kubectl version`):
* OS (e.g. from `/etc/os-release`):

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: manifests/kustomize/env/platform-agnostic-tekton-multi-user/kustomization.yaml [hunk 1]
```diff
@@ -15,13 +15,13 @@ commonLabels:
 images:
 - name: gcr.io/ml-pipeline/api-server
   newName: quay.io/aipipeline/apiserver
-  newTag: 2.1.0-rc0
+  newTag: 2.0.5
```
Comment (yhwang) [inline manifests/kustomize/env/platform-agnostic-tekton-multi-user/kustomization.yaml:18]: 2.1.0 to 2.0.5?

Comment (Tomcli) [reply manifests/kustomize/env/platform-agnostic-tekton-multi-user/kustomization.yaml:18]: I don't think kfp will release 2.1.0 anytime soon. Just want to cut a release with all the recent enhancement we have on Tekton 0.53

Comment (Tomcli) [reply manifests/kustomize/env/platform-agnostic-tekton-multi-user/kustomization.yaml:18]: We will use this release as a benchmark on the upcoming contributions to the Argo optimization.

**Review by yhwang — APPROVED** [review]
/lgtm

**Pull request merged**